srcTest/CompileCheckerTest.java

```diff
@@ -0,0 +1,94 @@
+        CompileChecker compileChecker = new CompileChecker();
+        String filePath = System.getProperty("user.dir") + "/srcTest/CompileCheckerTest.java";
+        File file = new File(filePath);
+        assertTrue(compileChecker.ensureFileIsJava(file));
+    }
+
+    @Test
+    public void canUseJavac() {
+        CompileChecker compileChecker = new CompileChecker();
+        List<String> arguments = new ArrayList<>();
+        String directory = System.getProperty("user.dir") + "/src";
+        File folder = new File(directory);
+        File[] listOfFiles = folder.listFiles();
+        for(int i = 0; i < listOfFiles.length; i++) {
+            if(listOfFiles[i].getName().contains(".java"))
+                arguments.add(listOfFiles[i].getPath());
+        }
+        boolean canCompile = compileChecker.createClassFiles(arguments.toArray(new String[arguments.size()]));
+        assertTrue(canCompile);
+    }
+
+    @Test
+    public void getAllFilesShouldReturnZero() {
+        CompileChecker compileChecker = new CompileChecker();
+        List<String> classes = compileChecker.getClassFiles();
+        for(int i = 0; i < classes.size(); i++) {
+            File file = new File(classes.get(i));
+            file.delete();
+        }
+        classes = compileChecker.getClassFiles();
+        assertTrue(classes.size() == 0);
+    }
+
+    @Test
+    public void ensureClassFilesAreGenerated() {
+        CompileChecker compileChecker = new CompileChecker();
+        List<String> classes = compileChecker.getClassFiles();
+        for(int i = 0; i < classes.size(); i++) {
+            File file = new File(classes.get(i));
+            file.delete();
+        }
+        List<String> arguments = new ArrayList<>();
+        String directory = System.getProperty("user.dir") + "/src";
+        File folder = new File(directory);
+        File[] listOfFiles = folder.listFiles();
+        for(int i = 0; i < listOfFiles.length; i++) {
+            if(listOfFiles[i].getName().contains(".java"))
+                arguments.add(listOfFiles[i].getPath());
+        }
+        compileChecker.createClassFiles(arguments.toArray(new String[arguments.size()]));
+        classes = compileChecker.getClassFiles();
+        assertTrue(classes.size() != 0);
+    }
+
+    @Test
+    public void buildJavaJarFile() {
+        CompileChecker compileChecker = new CompileChecker();
+        List<String> arguments = new ArrayList<>();
+        String directory = System.getProperty("user.dir") + "/src";
+        File folder = new File(directory);
+        File[] listOfFiles = folder.listFiles();
+        for(int i = 0; i < listOfFiles.length; i++) {
+            if(listOfFiles[i].getName().contains(".java"))
+                arguments.add(listOfFiles[i].getPath());
+        }
+        boolean canBuildJar = false;
+        boolean canCompile = compileChecker.createClassFiles(arguments.toArray(new String[arguments.size()]));
+        if(canCompile == true) {
```

Comment:
The presence of logical statements in unit tests is a code smell. My suggestion would be to create separate classes for these the createClassFiles method and the createJARFile method. That way, the return value can be mocked out to always return true. This would change the existing structure greatly.